### DISCUSSION TITLE
dark mode not working in production

### DISCUSSION THREAD

**Discussion by yudax42** [tailwindlabs/tailwindcss#4358] [Help]
Hello!
I have a problem using  dark mode, in development mode using Nuxtjs it's work fine I add class dark to HTML and it automatically switches to dark mode but in production after building it no longer works (after running npm run generate)
I see in chrome developer Elements that the class is added to the HTML tag, but the style is not applied to the concerned elements.
website Link: https://notion.yudax.dev/
"nuxt": "^2.15.3"
 "@nuxtjs/tailwindcss": "^4.0.1",
node: v15.14.0
Browser: Chrome
Thank you

------------------

Edit:
it's fixed now, i found that PurgeCss auto-remove non used classes in production and i allowed dark in config file

**Comment by rtkirill** [comment]
Thanks @yudax42! 
Wasted some time before found your solution 🙌

More details:

You should add safelist option to your tailwind config.

```
// tailwind.config.js Tailwindv2
module.exports = {
  purge: {
    content: ['./src/**/*.html'],

    // These options are passed through directly to PurgeCSS
    options: {
      safelist: ['dark']
    },
  },
  // ...
}
```

**Comment by ajaythakkar** [comment]
I have also the same problem now but safelist: ['dark'] is not working with v3.0.7 

Can you please help me?

**Reply by ajaythakkar** [reply]
This works for me :

`darkMode: ['variant', '&:not(.light *)'],`
